src/python/WMComponent/RucioInjector/RucioInjectorPoller.py

```diff
@@ -343,67 +347,72 @@ def deleteBlocks(self):
+
+            # Delete agent created rules locking the block
+            binds = []
+            logging.info("Going to delete %d block rules", len(blocksToDelete))
+            for block in blocksToDelete:
+                try:
+                    rules = self.rucio.listDataRules(block, scope=self.scope, account=self.rucioAcct)
+                except WMRucioException as exc:
+                    logging.warning("Unable to retrieve replication rules for block: %s. Will retry in the next cycle.", block)
+                else:
+                    if not rules:
+                        logging.info("Block rule for: %s has been deleted by previous cycles", block)
+                        binds.append({'DELETED': 1, 'BLOCKNAME': block})
+                        continue
+                    for rule in rules:
+                        deletedRules = 0
+                        if self.rucio.deleteRule(rule['id'], purgeReplicas=True):
+                            logging.info("Successfully deleted rule: %s, for block %s.", rule['id'], block)
+                            deletedRules += 1
+                        else:
+                            logging.warning("Failed to delete rule: %s, for block %s. Will retry in the next cycle.", rule['id'], block)
+                    if deletedRules == len(rules):
+                        binds.append({'DELETED': 1, 'BLOCKNAME': block})
+                        logging.info("Successfully deleted all rules for block %s.", block)
 
-        # Delete agent created rules locking the block
-        binds = []
-        logging.info("Going to delete %d block rules", len(blocksToDelete))
-        for block in blocksToDelete:
-            try:
-                rules = self.rucio.listDataRules(block, scope=self.scope, account=self.rucioAcct)
-            except WMRucioException as exc:
-                logging.warning("Unable to retrieve replication rules for block: %s. Will retry in the next cycle.", block)
-            else:
-                if not rules:
-                    logging.info("Block rule for: %s has been deleted by previous cycles", block)
-                    binds.append({'DELETED': 1, 'BLOCKNAME': block})
-                    continue
-                for rule in rules:
-                    deletedRules = 0
-                    if self.rucio.deleteRule(rule['id'], purgeReplicas=True):
-                        logging.info("Successfully deleted rule: %s, for block %s.", rule['id'], block)
-                        deletedRules += 1
-                    else:
-                        logging.warning("Failed to delete rule: %s, for block %s. Will retry in the next cycle.", rule['id'], block)
-                if deletedRules == len(rules):
-                    binds.append({'DELETED': 1, 'BLOCKNAME': block})
-                    logging.info("Successfully deleted all rules for block %s.", block)
 
+            self.markBlocksDeleted.execute(binds)
```

Comment:
It seems to me this piece needs to be one  level up, otherwise it would account only for blocks deleted  during the last slice. Need to double check that one  myself once again.

Reply:
no need to take actions on this one, I'll come back to this again later

Reply:
I think you are right. If I keep this here, it'll  work well if there are no issues. But, if the routine is interrupted at some point, the DB will be left outdated. I'll fix it.

Reply:
thanks @germanfgv